src/jit/gentree.h

```diff
@@ -1078,6 +1076,17 @@ struct GenTree
         }
     }
 
+    bool OperIs(genTreeOps oper)
+    {
+        return OperGet() == oper;
+    }
+
+    template <typename... T>
+    bool OperIs(genTreeOps oper, T... rest)
```

Comment:
Nice!

Out of curiosity, have you looked at the codegen for this sort of recursive variadic template when used with >2 arguments?